core/src/stack/inputs.rs

```diff
@@ -82,16 +82,15 @@ impl Serializable for StackInputs {
         // here and the one provided by default from winterfell.
 
         debug_assert!(self.values.len() <= u32::MAX as usize);
-        target.write_u32(self.values.len() as u32);
+        target.write_usize(self.values.len());
```

Comment:
Looking at this now, I think we should change the max number of stack inputs to `u16:MAX`.

Reply:
Do we want to change the total length of the stack to `u16::MAX`, or this change will affect only inputs and outputs?

Reply:
These changes affect only stack inputs/outputs.